File: MissingValue.py
# -*- coding: utf-8 -*-
"""
Created on Wed Apr 21 13:53:23 2021

@author: User
"""

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


claimants = pd.read_csv('F:/DataSets/claimants.csv')

# check for count of NA value
claimants.isna().sum()

##### Mean and Median imputer are used for numeric data. 

from sklearn.impute import SimpleImputer

##### Mean Imputer 
mean_imputer = SimpleImputer(missing_values=np.nan, strategy='mean')
claimants["CLMSEX"] = pd.DataFrame(mean_imputer.fit_transform(claimants[["CLMSEX"]]))
claimants["CLMSEX"].isna().sum()  

claimants["CLMINSUR"] = pd.DataFrame(mean_imputer.fit_transform(claimants[["CLMINSUR"]]))
claimants["CLMINSUR"].isna().sum()  

##### Median Imputer
median_imputer = SimpleImputer(missing_values=np.nan, strategy='median')
claimants["SEATBELT"] = pd.DataFrame(median_imputer.fit_transform(claimants[["SEATBELT"]]))
claimants["SEATBELT"].isna().sum() 

claimants["CLMAGE"] = pd.DataFrame(median_imputer.fit_transform(claimants[["CLMAGE"]]))
claimants["CLMAGE"].isna().sum() 

File: MissingValue.R

install.packages("readr")
library(readr) 

install.packages("missForest")
library(missForest)

claimants <- read.csv(file.choose())
sum(is.na(claimants)) 
summary(claimants)

claimants$CLMSEX[is.na(claimants$CLMSEX)] <- mean(claimants$CLMSEX, na.rm = TRUE)
sum(is.na(claimants$CLMSEX)) 

claimants$CLMINSUR[is.na(claimants$CLMINSUR)] <- mean(claimants$CLMINSUR, na.rm = TRUE)
sum(is.na(claimants$CLMINSUR)) 

claimants$SEATBELT[is.na(claimants$SEATBELT)] <- mean(claimants$SEATBELT, na.rm = TRUE)
sum(is.na(claimants$SEATBELT)) 

claimants$CLMAGE[is.na(claimants$CLMAGE)] <- mean(claimants$CLMAGE, na.rm = TRUE)
sum(is.na(claimants$CLMAGE)) 

